src/app/shared/background.model.ts

```diff
@@ -1,15 +1,44 @@
-import { Die, } from './die.model';
+import { Die } from './die.model';
 
 export class CharacterBackground {
-    constructor(
-    public number: number,
-    public name: string,
-    public backgroundText: string,
-    public qualities: string[],
-    public principleCategory: string,
-    public powerSourceDice: Die[],
-    public diceText: string,
-    public pageNumber: number,
-    public diceToAssign: Die[] = [],
-    ) {}
+    public number: number;
+    public name: string;
+    public backgroundText: string;
+    public qualities: string[];
+    public principleCategory: string;
+    public powerSourceDice: Die[] = [];
+    public diceText: string;
+    public pageNumber: number;
+    public diceToAssign: Die[] = [];
+
+    constructor(backgroundData: {
+        num: number;
+        name: string;
+        qualities: string[];
+        quality_dice: number[];
+        principle: string;
+        power_source_dice: number[];
+        page: number;
+        blurb: string;
+        background_text: string;
+        dice_text: string;
+    }) {
+        this.number = backgroundData.num;
```

Comment:
in a previous comment, I mentioned it would be useful to have the class and its constructor object have the same interface, so that the properties naming and typing align.  If we did have that, this section could be made much simpler by using [Object.assign](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign)

replace
```
        this.number = backgroundData.num;
        this.name = backgroundData.name;
        this.backgroundText = backgroundData.background_text;
        this.qualities = backgroundData.qualities;
        this.principleCategory = backgroundData.principle;
        this.diceText = backgroundData.dice_text;
        this.pageNumber = backgroundData.page;
```
with this
```
       Object.assign(this, backgroundData); // take all props on backgroundData and assign them to this instance
```

Reply:
this was exactly what I had wanted to do with it, but couldn't figure out how to find the way to do that.   I'm going to look into the documentation on that so I can get a better handle on that as well.